apps/web/src/server/applications/case/project-team/applications-project-team.service.js

```diff
@@ -40,3 +62,58 @@ export const searchProjectTeamMembers = async (searchTerm, ADToken, pageNumber)
 
 	return response;
 };
+
+/**
+ * Retrieve the case team member info stored in the db having the given userid
+ *
+ * @param {number} caseId
+ * @param {string} userId
+ * @returns {Promise<{projectTeamMember: {userId: string, role: string} , errors: ValidationErrors}>}
+ */
+export const getProjectTeamMemberById = async (caseId, userId) => {
+	let response;
+
+	try {
+		const projectTeamMember = await get(`applications/${caseId}/project-team/${userId}`);
+
+		response = { projectTeamMember };
+	} catch (/** @type {*} */ error) {
+		pino.error(`[API] ${error?.response?.body?.error?.code || 'Unknown error'}`);
+
+		response = new Promise((resolve) => {
+			resolve({ errors: { query: 'An error occurred, please try again later' } });
+		});
```

Comment:
`Promise.resolve` should do the same.

```suggestion
		response = Promise.resolve({ errors: { query: 'An error occurred, please try again later' } });
```

or maybe even just using the value, not sure if it needs to be a promise.

```suggestion
		response = { errors: { query: 'An error occurred, please try again later' } };
```

Reply:
this syntax is a way to trick JSDocs.
For some reason the RequestHandler functions of Express in this project expect the errors to be in the format 
``import('@pins/express').ValidationErrors``: a very complicated interface very different from the `Record<string, 'ERROR MESSAGE'>` that you would expect.

I'm not sure how to fix it, so returning a new Promise.resolve just confuses the type checking and allows me to return a reasonable error response without typing errors. Setting the value as `any` would probably help, but I try not to do it.

In this case, I did some changes to allow the Record<string,string> and the ValidationErrors to co-exist, so I could use your suggestion. However this same code is used everywhere else in the frontend.